From b36fbd6277c54888e2c7ce4e82a0bfba1029cd77 Mon Sep 17 00:00:00 2001
From: Michal Suchanek <msuchanek@suse.de>
Date: Thu, 7 May 2020 19:39:05 +0200
Subject: [PATCH] libnvdimm: cover up struct nvdimm changes

References: bsc#1171742
Patch-mainline: never, kabi

Functions using this structure are exported but the layout is declared
in nd-core.h header internal to nvdimm core. The size and layout of the
structure has changed but out-of-tree users should not be able to tell.

Signed-off-by: Michal Suchanek <msuchanek@suse.de>
---
 drivers/nvdimm/nd-core.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/nvdimm/nd-core.h b/drivers/nvdimm/nd-core.h
index 52903a76f434..ae935194c3df 100644
--- a/drivers/nvdimm/nd-core.h
+++ b/drivers/nvdimm/nd-core.h
@@ -39,8 +39,13 @@ struct nvdimm {
 	const char *dimm_id;
 	struct {
 		const struct nvdimm_security_ops *ops;
+#ifndef __GENKSYMS__
 		unsigned long flags;
 		unsigned long ext_flags;
+#else
+		enum nvdimm_security_state state;
+		enum nvdimm_security_state ext_state;
+#endif
 		unsigned int overwrite_tmo;
 		struct kernfs_node *overwrite_state;
 	} sec;
-- 
2.26.2

